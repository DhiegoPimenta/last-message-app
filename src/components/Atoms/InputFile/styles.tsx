import styled, { css } from 'styled-components'

export const DragDropFile = styled.label`
    ${({ theme }) => css`
        border: 4px dashed ${theme.colors.danger};
        background-size: cover;
        cursor: pointer;
        height: 33rem;
        width: 100%;
        border-radius: ${theme.border.radius};

        display: flex;
        justify-content: center;
        align-items: center;

        input {
            display: none;
        }

        .has-thumbnail {
            border: 0;
        }

        .has-thumbnail img {
            display: none;
        }
    `}
`
