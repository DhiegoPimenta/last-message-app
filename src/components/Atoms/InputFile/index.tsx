import { ChangeEvent, useMemo } from 'react'
import Text from '../Text'
import * as S from './styles'

export interface InputFileProps
    extends React.ButtonHTMLAttributes<HTMLInputElement> {
    thumbnail: any
    setThumbnail: any
}
const InputFile = ({ thumbnail, setThumbnail }: InputFileProps) => {
    const preview = useMemo(() => {
        return thumbnail ? URL.createObjectURL(thumbnail) : null
    }, [thumbnail])

    return (
        <S.DragDropFile
            id="thumbnail"
            style={{ backgroundImage: `url(${preview})` }}
            className={thumbnail ? 'has-thumbnail' : ''}
        >
            <input
                type="file"
                onChange={(event: ChangeEvent<HTMLInputElement>) =>
                    setThumbnail(event.target.files?.[0])
                }
            />
            <Text fullWidth color="gray">
                Adicione seu video ou imagem
            </Text>
        </S.DragDropFile>
    )
}
export default InputFile
