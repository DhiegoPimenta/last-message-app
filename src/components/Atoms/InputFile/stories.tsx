import { Story, Meta } from '@storybook/react/types-6-0'

import InputFile, { InputFileProps } from '.'

export default {
    title: 'Atoms/InputFile',
    component: InputFile,
} as Meta

export const Basic: Story<InputFileProps> = (args) => <InputFile {...args} />
Basic.args = {}
