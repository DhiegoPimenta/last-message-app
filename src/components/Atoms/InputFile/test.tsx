import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'
import InputFile from '.'

describe('<InputFile/>', () => {
    it('Should be render InputFile', () => {
        const { container } = renderWithTheme(<InputFile />)

        expect(
            screen.getByText(/Adicione seu video ou imagem/i)
        ).toBeInTheDocument()
        expect(container.firstChild).toMatchSnapshot()
    })
})
