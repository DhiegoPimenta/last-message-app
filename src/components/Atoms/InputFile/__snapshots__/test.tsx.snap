// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputFile/> Should be render InputFile 1`] = `
.c1 {
  color: #CECECE;
  font-size: 2.4rem;
  width: 100%;
  display: block;
  font-weight: 400;
  text-align: center;
}

.c0 {
  border: 4px dashed #FBA9B0;
  background-size: cover;
  cursor: pointer;
  height: 33rem;
  width: 100%;
  border-radius: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 input {
  display: none;
}

.c0 .has-thumbnail {
  border: 0;
}

.c0 .has-thumbnail img {
  display: none;
}

<label
  class="c0"
  id="thumbnail"
  style="background-image: url(null);"
>
  <input
    type="file"
  />
  <p
    class="c1"
    color="gray"
  >
    Adicione seu video ou imagem
  </p>
</label>
`;
